Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp

```diff
@@ -620,7 +620,8 @@ void WebKitProtocolHandler::handleGPU(WebKitURISchemeRequest* request)
         "console.log(JSON.stringify(%s, null, 4));"
         "}</script>", infoAsString.utf8().data());
 
-    g_string_append(html, "</head><body>");
+    // dump to JS console at startup to make easier getting this info on remote/headless/virtual displays
+    g_string_append(html, "</head><body onload=\"sendToConsole()\">");
```

Comment:
Do we really want to do this unconditionally? I think we can do this only when requested using the request path, and we could add more actions, for example

* webkit://gpu/json/console: send the json to the console like this patch does.
* webkit://gpu/json/stdout: prints the json in stdout
* webkit://gpu/json/clipboard: writes the json in the clipboard

These options are useful not only for embedded but also when debugging rendering issues where nothing is rendered.

Reply:
I would vote to keep this simple, the dumping does not hurt performance and we have less things to remember/document.

I'm not opposed to it though...

Reply:
Another point of view:

Dumping to console is enough, we already have a flag to dump console to stdout, and command line tools to copy to clipboard.

Reply:
It's not because of performance, in the gpu page info performance doesn't matter. But what's the point of having a button to send the json to the console if we have already sent it unconditionally on load? Having the info in stout directly is useful too, with this patch we need to enable the setting to send console messages to stout too, which I don't think it's simpler than just typing `webkit://gpu/json/stdout`

Reply:
> Another point of view:
> 
> Dumping to console is enough, we already have a flag to dump console to stdout, and command line tools to copy to clipboard.

what flag do you mean? and what command line to copy to clipboard?

Reply:
We have a feature flag, that we can pass as command line argument to dump JS console to stdout - I was referring to that one.

I did not think of the button, in my opinion, it can be removed.

There is no WebKit specific tool to copy to clipboard but e.g. wl-copy.

Reply:
That command line option is present in cog and MiniBrowser, not in all apps using WPE, loading `webkit://gpu/json/stdout `would always be available not matter what browser app is used. Copying to clipboard is specific to GTK port, so `webkit://gpu/json/clipboard` would also be specific to GTK. It has happened many times in gtk port that people reporting nothing is rendered load `webkit://gpu` and click where the button is supposed to be to get the result in the clipboard. In those cases it would be a lot easier to just load a different url, instead of having to enable a setting and reload

Reply:
> In those cases it would be a lot easier to just load a different url, instead of having to enable a setting and reload

But enabling the setting is also needed when loading the different url, isn't it?

Reply:
Not when using `webkit://gpu/json/stdout`

Reply:
I see the tricky part is formatting the json like stringify does... So, let's keep it this way, but let's remove the button then, since it's always sent to the console. Also, even if existing issue, we can avoid the utf8 conversion made twice by doing:

```
auto infoAsString = jsonObject->toJSONString().utf8();
```

And then using the CString in both `copyAsJSON` and `sendToConsole`. In the case of WPE we don't need to generate the `copyAsJSON` since we don't add the button.

Reply:
The absence of the command line option in non-cog/non-MiniBrowser scenarios convinced me, good point @carlosgcampos.